Demo/Sources/YouTube/PlayerItem.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  Copyright (c) SRG SSR. All rights reserved.
+//
+//  License information is available from the LICENSE file.
+//
+
+import Combine
+import Player
+import UIKit
+import YouTubeKit
+
+extension PlayerItem {
+    static func youTube(videoId: String) -> Self {
+        let youTubePublisher = imagePublisher(videoId: videoId)
+            .prepend(UIImage())
+            .map { image in
+                urlPublisher(videoId: videoId)
+                    .map { url in
+                        Asset.simple(url: url, metadata: YouTubeMetadata(image: image))
+                    }
+            }
+            .switchToLatest()
+            .eraseToAnyPublisher()
+
+        return self.init(publisher: youTubePublisher)
```

Comment:
We can probably rewrite this as a `CombineLatest` publisher since both pipelines are independent.

Reply:
To use a `CombineLastest` the error of the publisher `A` has to be equal to the error of the publisher `B`.
It's not our case, the `urlPublisher` has an `YouTubeError` and the `imagePublisher` has no error.

As discussed, we can use `setFailureType` ☺️